great_expectations/datasource/fluent/snowflake_datasource.py

```diff
@@ -26,4 +37,59 @@ class SnowflakeDatasource(SQLDatasource):
     """
 
     type: Literal["snowflake"] = "snowflake"  # type: ignore[assignment]
-    connection_string: Union[ConfigStr, SnowflakeDsn]
+    connection_string: Optional[Union[ConfigStr, SnowflakeDsn]] = None
+    # connect_args
+    account: Optional[str] = None
+    user: Optional[str] = None
+    password: Optional[str] = None
```

Comment:
```suggestion
    password: Optional[Union[ConfigStr, str]] = None
```
If we want to support config substitution for this value then `ConfigStr` must be part of the annotation, so that pydantic will attempt to coerce the value into a `ConfigStr`